gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_catalog/TopologyCatalog.java

```diff
@@ -232,6 +232,15 @@ public void switchMetricContext(MetricContext context) {
     }
   }
 
+  @Override
+  public int getSize() {
+    try {
+      return specStore.getSize();
+    } catch (IOException e) {
+      throw new RuntimeException("Cannot retrieve number of specs from Spec store", e);
+    }
+  }
```

Comment:
[thought for later] literal duplication (in each containing class):  perhaps explore a `CheckedExceptionWrappingSpecStore`, basically a Decorator for centralizing this?  doing so lends similar benefit as the `InstrumentedSpecStore` by encapsulating timing in one place (although based on composition rather than impl inheritance, as ISS is).